Importing libraries

In [1]:
import pandas as pd
import numpy as np

Loading data

In [2]:
df=pd.read_csv("Airbnb_Rome.csv")

In [3]:
df.head()

Airbnb_type       City Price_before_sale price_after_sale  Rates  \
0           Room  Tuscolano         ج.م10,312         ج.م7,147   4.64   
1           Loft      Trevi         ج.م73,640        ج.م32,551   4.77   
2      Apartment      Trevi         ج.م77,592        ج.م63,648   4.80   
3      Apartment      Celio        ج.م166,053       ج.م137,055   5.00   
4  Vacation home       Rome         ج.م60,379        ج.م36,644   4.88   

   Reviews_Num    Host_name    Host_Profession  
0           14       Daniel  Social technician  
1          163     Federico              Audio  
2          716  Juan Carlos           Musician  
3           69      Valeria          Freelance  
4           98            G        Hosting for

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 298 entries, 0 to 297
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Airbnb_type        298 non-null    object 
 1   City               298 non-null    object 
 2   Price_before_sale  298 non-null    object 
 3   price_after_sale   298 non-null    object 
 4   Rates              298 non-null    float64
 5   Reviews_Num        298 non-null    int64  
 6   Host_name          295 non-null    object 
 7   Host_Profession    295 non-null    object 
dtypes: float64(1), int64(1), object(6)
memory usage: 18.8+ KB


*Host name and profession name have null values which maybe unspecified so we can fillna with unknown.
*price column needs to be converted into int.

In [13]:
df.fillna("unknown",inplace=True)

In [6]:
import re
def clean_egyptian_price(price_str):
    return pd.to_numeric(
        price_str.str.replace('ج.م', '', regex=False)
                  .str.replace(',', '', regex=False)
                  .str.strip()
    )

# Apply to your columns
df['Price_before_sale'] = clean_egyptian_price(df['Price_before_sale'])
df['price_after_sale'] = clean_egyptian_price(df['price_after_sale'])


In [ ]:
df.describe()
#prices have right skweness

Price_before_sale  price_after_sale       Rates  Reviews_Num  \
count         298.000000        298.000000  298.000000   298.000000   
mean        47254.986577      34739.570470    4.875772   163.969799   
std         37816.670837      26227.976534    0.120402   187.076883   
min         10312.000000       7147.000000    4.490000     3.000000   
25%         29315.000000      22646.000000    4.800000    15.250000   
50%         36436.500000      26575.000000    4.900000    97.500000   
75%         52238.000000      36644.000000    5.000000   245.500000   
max        275627.000000     172241.000000    5.000000   833.000000   

       Dicount_Amount   Discount%  
count      298.000000  298.000000  
mean     12515.416107   24.654362  
std      14010.836579   11.623933  
min       1157.000000    8.000000  
25%       5223.000000   15.000000  
50%       8557.000000   22.000000  
75%      15804.000000   32.000000  
max     103961.000000   56.000000

In [9]:
df.duplicated().sum()

np.int64(7)

In [15]:
df.drop_duplicates()

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
0             Room  Tuscolano              10312              7147   4.64   
1             Loft      Trevi              73640             32551   4.77   
2        Apartment      Trevi              77592             63648   4.80   
3        Apartment      Celio             166053            137055   5.00   
4    Vacation home       Rome              60379             36644   4.88   
..             ...        ...                ...               ...    ...   
293      Apartment       Rome              58686             44898   5.00   
294           Room       Rome              69365             49867   4.72   
295      Apartment       Rome              48382             39170   4.67   
296      Apartment       Rome              54534             49758   4.77   
297           Loft       Rome              23760             21393   4.73   

     Reviews_Num    Host_name    Host_Profession  
0             14       Daniel  Social technician  
1            163     Federico              Audio  
2            716  Juan Carlos           Musician  
3             69      Valeria          Freelance  
4             98            G        Hosting for  
..           ...          ...                ...  
293            4      Unknown            Unknown  
294           58      Unknown            Unknown  
295            3      unknown            unknown  
296           94      unknown            unknown  
297          110      unknown            unknown  

[291 rows x 8 columns]

#Feature Engineering

In [16]:
df["Dicount_Amount"]=df["Price_before_sale"]-df["price_after_sale"]

In [23]:
df["Discount%"]=((df["Dicount_Amount"]/df["Price_before_sale"])*100).round(0)

In [24]:
df["Discount%"]

0      31.0
1      56.0
2      18.0
3      17.0
4      39.0
       ... 
293    23.0
294    28.0
295    19.0
296     9.0
297    10.0
Name: Discount%, Length: 298, dtype: float64

In [25]:
df

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
0             Room  Tuscolano              10312              7147   4.64   
1             Loft      Trevi              73640             32551   4.77   
2        Apartment      Trevi              77592             63648   4.80   
3        Apartment      Celio             166053            137055   5.00   
4    Vacation home       Rome              60379             36644   4.88   
..             ...        ...                ...               ...    ...   
293      Apartment       Rome              58686             44898   5.00   
294           Room       Rome              69365             49867   4.72   
295      Apartment       Rome              48382             39170   4.67   
296      Apartment       Rome              54534             49758   4.77   
297           Loft       Rome              23760             21393   4.73   

     Reviews_Num    Host_name    Host_Profession  Dicount_Amount  Discount%  
0             14       Daniel  Social technician            3165       31.0  
1            163     Federico              Audio           41089       56.0  
2            716  Juan Carlos           Musician           13944       18.0  
3             69      Valeria          Freelance           28998       17.0  
4             98            G        Hosting for           23735       39.0  
..           ...          ...                ...             ...        ...  
293            4      Unknown            Unknown           13788       23.0  
294           58      Unknown            Unknown           19498       28.0  
295            3      unknown            unknown            9212       19.0  
296           94      unknown            unknown            4776        9.0  
297          110      unknown            unknown            2367       10.0  

[298 rows x 10 columns]

Analysis

Top 10 Most Expensive Airbnb Before Sale

In [26]:
df.sort_values(by='Price_before_sale', ascending=False).head(10)

Airbnb_type             City  Price_before_sale  price_after_sale  Rates  \
121   Apartment  Castro Pretorio             275627            171666   4.95   
37    Apartment  Castro Pretorio             275627            171666   4.95   
163   Apartment  Castro Pretorio             275627            171666   4.95   
79    Apartment  Castro Pretorio             275627            171666   4.95   
192   Apartment             Rome             235486            172241   4.79   
254   Apartment             Rome             188036            153322   4.85   
3     Apartment            Celio             166053            137055   5.00   
45    Apartment            Celio             166053            137055   5.00   
129   Apartment            Celio             166053            137055   5.00   
87    Apartment            Celio             166053            137055   5.00   

     Reviews_Num  Host_name    Host_Profession  Dicount_Amount  Discount%  
121          297   Ludovica            Painter          103961       38.0  
37           297  Benedetto  Artistic director          103961       38.0  
163          297     Mugiko        Hospitality          103961       38.0  
79           297      Piero  Software engineer          103961       38.0  
192           58    Unknown            Unknown           63245       27.0  
254           13    Unknown            Unknown           34714       18.0  
3             69    Valeria          Freelance           28998       17.0  
45            69          G        Hosting for           28998       17.0  
129           69      Royal        Hosting for           28998       17.0  
87            69     Lilith       Secret agent           28998       17.0

Top 10 Cheapest Airbnb Before Sale

In [27]:
df.sort_values(by='Price_before_sale', ascending=True).head(10)

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
0          Room  Tuscolano              10312              7147   4.64   
42         Room  Tuscolano              10312              7147   4.64   
84         Room  Tuscolano              10312              7147   4.64   
126        Room  Tuscolano              10312              7147   4.64   
90         Room  Tuscolano              14693             13536   5.00   
48         Room  Tuscolano              14693             13536   5.00   
132        Room  Tuscolano              14693             13536   5.00   
6          Room  Tuscolano              14693             13536   5.00   
265       Condo       Rome              18030             14790   4.97   
208       Condo       Rome              18341             16605   4.67   

     Reviews_Num    Host_name    Host_Profession  Dicount_Amount  Discount%  
0             14       Daniel  Social technician            3165       31.0  
42            14     Federico              Audio            3165       31.0  
84            14  Juan Carlos           Musician            3165       31.0  
126           14      Valeria          Freelance            3165       31.0  
90           192       Giulia           Employee            1157        8.0  
48           192    Francesco        Hospitality            1157        8.0  
132          192        Elena        Hosting for            1157        8.0  
6            192        Royal        Hosting for            1157        8.0  
265           59      Unknown            Unknown            3240       18.0  
208          147      Unknown            Unknown            1736        9.0

Top 10 Cheapest Airbnb After Sale

In [28]:
df.sort_values(by='price_after_sale', ascending=True).head(10)

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
0          Room  Tuscolano              10312              7147   4.64   
42         Room  Tuscolano              10312              7147   4.64   
84         Room  Tuscolano              10312              7147   4.64   
126        Room  Tuscolano              10312              7147   4.64   
169       Condo       Rome              18878             11357   4.83   
145        Room  Tuscolano              18878             11357   4.80   
103        Room  Tuscolano              18878             11357   4.80   
61         Room  Tuscolano              18878             11357   4.80   
19         Room  Tuscolano              18878             11357   4.80   
241       Condo       Rome              21739             11861   4.56   

     Reviews_Num    Host_name    Host_Profession  Dicount_Amount  Discount%  
0             14       Daniel  Social technician            3165       31.0  
42            14     Federico              Audio            3165       31.0  
84            14  Juan Carlos           Musician            3165       31.0  
126           14      Valeria          Freelance            3165       31.0  
169          127      Unknown            Unknown            7521       40.0  
145          260            G        Hosting for            7521       40.0  
103          260      Daniele        Hosting for            7521       40.0  
61           260       Sergio             Artist            7521       40.0  
19           260       Mugiko        Hospitality            7521       40.0  
241          139      Unknown            Unknown            9878       45.0

Top 10 Most Expensive Airbnb After Sale

In [29]:
df.sort_values(by='price_after_sale', ascending=False).head(10)

Airbnb_type             City  Price_before_sale  price_after_sale  Rates  \
192   Apartment             Rome             235486            172241   4.79   
121   Apartment  Castro Pretorio             275627            171666   4.95   
37    Apartment  Castro Pretorio             275627            171666   4.95   
163   Apartment  Castro Pretorio             275627            171666   4.95   
79    Apartment  Castro Pretorio             275627            171666   4.95   
254   Apartment             Rome             188036            153322   4.85   
3     Apartment            Celio             166053            137055   5.00   
45    Apartment            Celio             166053            137055   5.00   
129   Apartment            Celio             166053            137055   5.00   
87    Apartment            Celio             166053            137055   5.00   

     Reviews_Num  Host_name    Host_Profession  Dicount_Amount  Discount%  
192           58    Unknown            Unknown           63245       27.0  
121          297   Ludovica            Painter          103961       38.0  
37           297  Benedetto  Artistic director          103961       38.0  
163          297     Mugiko        Hospitality          103961       38.0  
79           297      Piero  Software engineer          103961       38.0  
254           13    Unknown            Unknown           34714       18.0  
3             69    Valeria          Freelance           28998       17.0  
45            69          G        Hosting for           28998       17.0  
129           69      Royal        Hosting for           28998       17.0  
87            69     Lilith       Secret agent           28998       17.0

Airbnb with Highest Discount Amount

In [30]:
df[df['Dicount_Amount'] == df['Dicount_Amount'].max()]

Airbnb_type             City  Price_before_sale  price_after_sale  Rates  \
37    Apartment  Castro Pretorio             275627            171666   4.95   
79    Apartment  Castro Pretorio             275627            171666   4.95   
121   Apartment  Castro Pretorio             275627            171666   4.95   
163   Apartment  Castro Pretorio             275627            171666   4.95   

     Reviews_Num  Host_name    Host_Profession  Dicount_Amount  Discount%  
37           297  Benedetto  Artistic director          103961       38.0  
79           297      Piero  Software engineer          103961       38.0  
121          297   Ludovica            Painter          103961       38.0  
163          297     Mugiko        Hospitality          103961       38.0

In [31]:
df[df['Dicount_Amount'] == df['Dicount_Amount'].min()]

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
6          Room  Tuscolano              14693             13536    5.0   
48         Room  Tuscolano              14693             13536    5.0   
90         Room  Tuscolano              14693             13536    5.0   
132        Room  Tuscolano              14693             13536    5.0   

     Reviews_Num  Host_name Host_Profession  Dicount_Amount  Discount%  
6            192      Royal     Hosting for            1157        8.0  
48           192  Francesco     Hospitality            1157        8.0  
90           192     Giulia        Employee            1157        8.0  
132          192      Elena     Hosting for            1157        8.0

Number of Airbnb Listings Per City

In [32]:
df['City'].value_counts()

City
Rome               103
Tuscolano           39
Prati               23
Monti               18
Castro Pretorio     15
Trionfale           13
Trevi               12
della Vittoria       8
Aurelio              8
Don Bosco            7
Trastevere           6
Portuense            6
Celio                5
Parione              5
Trieste              5
Ostiense             5
Ludovisi             4
Testaccio            4
Campo Marzio         4
Frascati             3
Ponte Mammolo        1
Primavalle           1
Gianicolense         1
Piana del Sole       1
Pietralata           1
Name: count, dtype: int64

Airbnb with Lowest Rate

In [33]:
df[df['Rates'] == df['Rates'].min()]

Airbnb_type       City  Price_before_sale  price_after_sale  Rates  \
219   Apartment  Don Bosco              25531             23308   4.49   

     Reviews_Num Host_name Host_Profession  Dicount_Amount  Discount%  
219           61   Unknown         Unknown            2223        9.0

Airbnb with Highest Rate

In [34]:
df[df['Rates'] == df['Rates'].max()]

Airbnb_type            City  Price_before_sale  price_after_sale  Rates  \
3     Apartment           Celio             166053            137055    5.0   
6          Room       Tuscolano              14693             13536    5.0   
10    Apartment       Portuense              35541             22646    5.0   
13         Room       Don Bosco              32359             23802    5.0   
15    Apartment           Prati              35316             23769    5.0   
..          ...             ...                ...               ...    ...   
248        Home            Rome              43081             29712    5.0   
253   Apartment        Frascati              33284             28193    5.0   
255       Condo            Rome              29315             24826    5.0   
266        Home  Piana del Sole              34697             31362    5.0   
293   Apartment            Rome              58686             44898    5.0   

     Reviews_Num Host_name Host_Profession  Dicount_Amount  Discount%  
3             69   Valeria       Freelance           28998       17.0  
6            192     Royal     Hosting for            1157        8.0  
10             9    Lilith    Secret agent           12895       36.0  
13             4  Isabella            Chef            8557       26.0  
15             4     Annie      Researcher           11547       33.0  
..           ...       ...             ...             ...        ...  
248            4   Unknown         Unknown           13369       31.0  
253            3   Unknown         Unknown            5091       15.0  
255           22   Unknown         Unknown            4489       15.0  
266           13   Unknown         Unknown            3335       10.0  
293            4   Unknown         Unknown           13788       23.0  

[80 rows x 10 columns]

Mean of Price Before and After Sale

In [35]:
mean_before = df['Price_before_sale'].mean()
mean_after = df['price_after_sale'].mean()
print(f"Mean before sale: {mean_before:.2f}")
print(f"Mean after sale: {mean_after:.2f}")

Mean before sale: 47254.99
Mean after sale: 34739.57


Count of Each Host Profession

In [36]:
df['Host_Profession'].value_counts()

Host_Profession
Unknown                 131
Hosting for              40
Hospitality              20
Artistic director        12
Artist                    8
Illustrator               8
Secret agent              8
Architect                 8
Employee                  4
Audio                     4
Musician                  4
Freelance                 4
Social technician         4
Music organization        4
Chef                      4
Researcher                4
Television host           4
Performer                 4
Inventory specialist      4
Entertainment             4
Filmmaker                 4
Software engineer         4
Painter                   4
unknown                   3
Name: count, dtype: int64

EDA

In [39]:
import matplotlib.pyplot as plt
import plotly.express as px
import seaborn as sns
import plotly.graph_objects as go
from plotly.subplots import make_subplots


Histogram for prices right skwness pridected some Airbnb are way expensive

In [40]:

fig = make_subplots(rows=1, cols=2, subplot_titles=("Price Before Sale", "Price After Sale"))
fig.add_trace(
    go.Histogram(x=df['Price_before_sale'], nbinsx=30, name='Before Sale', marker_color='indigo'),
    row=1, col=1
)
fig.add_trace(
    go.Histogram(x=df['price_after_sale'], nbinsx=30, name='After Sale', marker_color='teal'),
    row=1, col=2
)
fig.update_layout(
    title_text="Airbnb Price Distribution: Before vs. After Sale",
    showlegend=False,
    height=500,
    width=1000,
    bargap=0.2
)

fig.show()

In [52]:


fig = px.bar(
    city_counts.sort_values(by='Count', ascending=False),
    x='City',
    y='Count',
    title='Number of Airbnb Listings per City in Rome',
    text='Count',
    color='City',
    color_discrete_sequence=px.colors.qualitative.Vivid
)

fig.update_layout(
    xaxis_title='City',
    yaxis_title='Number of Listings',
    showlegend=False
)

fig.show()

In [53]:

df['Rounded_Rate'] = df['Rates'].round(1)


rate_counts = df['Rounded_Rate'].value_counts().reset_index()
rate_counts.columns = ['Rate', 'Count']
rate_counts = rate_counts.sort_values(by='Rate')


fig = px.bar(
    rate_counts,
    x='Rate',
    y='Count',
    title='Number of Airbnb Listings by Rating',
    text='Count',
    color='Rate',
    color_continuous_scale='Viridis'
)

fig.update_layout(
    xaxis_title='Rating',
    yaxis_title='Number of Listings',
    showlegend=False
)

fig.show()

In [65]:
import plotly.express as px

fig = px.violin(
    df,
    x='Airbnb_type',
    y='Price_before_sale',
    color='Airbnb_type',
    box=True,  # Includes box plot inside
    points='outliers',  
    title='Price Distribution by Airbnb Type',
    color_discrete_sequence=px.colors.qualitative.Set1
)

fig.update_layout(xaxis_title='Airbnb Type', yaxis_title='Price Before Sale')
fig.show()

In [66]:

df['Rounded_Rate'] = df['Rates'].round(1)

price_per_rate = df.groupby('Rounded_Rate')['Price_before_sale'].mean().reset_index()
price_per_rate.columns = ['Rate', 'Avg_Price']


fig = px.bar(
    price_per_rate,
    x='Rate',
    y='Avg_Price',
    title='Average Airbnb Price Before Sale per Rating',
    text='Avg_Price',
    color='Rate',
    color_continuous_scale='Blues'
)

fig.update_layout(
    xaxis_title='Rating',
    yaxis_title='Average Price Before Sale',
    showlegend=False
)

fig.show()

Pivot tables

In [67]:
pivot_avg_price = df.pivot_table(values=['Price_before_sale', 'price_after_sale'], index='City', aggfunc='mean')


In [68]:
pivot_avg_discount = df.pivot_table(values=['Dicount_Amount', 'Discount%'], index='City', aggfunc='mean')


In [69]:
pivot_city_type = df.pivot_table(index='City', columns='Airbnb_type', aggfunc='size', fill_value=0)


In [70]:
pivot_avg_rating = df.pivot_table(values='Rates', index='City', aggfunc='mean')


In [71]:
pivot_prof_count = df.pivot_table(index='Host_Profession', aggfunc='size').to_frame(name='Count')


In [72]:
pivot_reviews_by_prof = df.pivot_table(values='Reviews_Num', index='Host_Profession', aggfunc='mean')


In [73]:
pivot_discount_by_prof = df.pivot_table(values='Discount%', index='Host_Profession', aggfunc='mean')


In [74]:
pivot_rating_by_type = df.pivot_table(values='Rates', index='Airbnb_type', aggfunc='mean')


In [75]:
pivot_price_by_rate = df.pivot_table(values=['Price_before_sale', 'price_after_sale'], index='Rates', aggfunc='mean')


In [76]:
pivot_city_type_price = df.pivot_table(values=['Price_before_sale', 'Dicount_Amount'], index=['City', 'Airbnb_type'], aggfunc='mean')

In [77]:


# Write them to Excel
with pd.ExcelWriter('airbnb_pivot_tables.xlsx') as writer:
    pivot_avg_price.to_excel(writer, sheet_name='Avg_Price_by_City')
    pivot_avg_discount.to_excel(writer, sheet_name='Avg_Discount_by_City')
    pivot_city_type.to_excel(writer, sheet_name='Listings_by_City_Type')
    pivot_avg_rating.to_excel(writer, sheet_name='Avg_Rating_by_City')
    pivot_prof_count.to_excel(writer, sheet_name='Host_Profession_Count')
    pivot_reviews_by_prof.to_excel(writer, sheet_name='Reviews_by_Profession')
    pivot_discount_by_prof.to_excel(writer, sheet_name='Discount_by_Profession')
    pivot_rating_by_type.to_excel(writer, sheet_name='Rating_by_Type')
    pivot_price_by_rate.to_excel(writer, sheet_name='Price_by_Rating')
    pivot_city_type_price.to_excel(writer, sheet_name='City_Type_Price')



Summary

By analyzing Airbnb available in Rome in August we got that:


1 Prices have right skweness due to some airbnb being so expensive other than others


2 Rome has the most Airbnb other than other cities

3 rate 5/5 is the highest in dataset

4 Aprtments are the most available during this month

5 rate 5 and 4.5 has the highest count and prices 